### PR TITLE
Add Plutus compatible Json instance for Types.Scripts

### PR DESCRIPTION
**Pull request by bladyjoker**

### CODE REVIEW

File: src/Plutus/Types/CurrencySymbol.purs [hunk 1]
```diff
@@ -10,15 +10,25 @@ module Plutus.Types.CurrencySymbol
 
 import Prelude
 
+import Contract.Prelude (Either(..))
+import Data.Argonaut
+  ( class DecodeJson
+  , class EncodeJson
+  , JsonDecodeError(..)
```
Comment (ngua) [inline src/Plutus/Types/CurrencySymbol.purs:17]: Usually we use explicit constructor imports

File: src/Types/Scripts.purs [hunk 2]
```diff
@@ -10,7 +10,17 @@ module Types.Scripts
 
 import Prelude
 
-import Data.Argonaut (class DecodeJson, class EncodeJson)
+import Contract.Prelude (Either(..))
+import Data.Argonaut
+  ( class DecodeJson
+  , class EncodeJson
+  , JsonDecodeError(..)
```
Comment (ngua) [inline src/Types/Scripts.purs:17]: Same here

File: src/Types/Scripts.purs [hunk 3]
```diff
@@ -10,7 +10,17 @@ module Types.Scripts
 
 import Prelude
 
-import Data.Argonaut (class DecodeJson, class EncodeJson)
+import Contract.Prelude (Either(..))
```
Comment (ngua) [inline src/Types/Scripts.purs:13]: We should avoid importing `Contract` modules internally

**Review by ngua — CHANGES_REQUESTED** [review]
The changes to the instances look fine, but a few things we need to do before merging this:
- remove the `Contract` imports
- update the examples to account for the changes to the JSON instances

File: src/Plutus/Types/CurrencySymbol.purs (hunk 1 above)
Comment (bladyjoker) [reply src/Plutus/Types/CurrencySymbol.purs:17]: IDE does this for some reason like this, terrible I agree, fixing

File: src/Types/Scripts.purs (hunk 3 above)
Comment (bladyjoker) [reply src/Types/Scripts.purs:13]: How did that get here, I guess IDE being very smoort again

**Review by ngua — APPROVED** [review]

**Pull request merged**